go/vt/orchestrator/logic/topology_recovery.go

```diff
@@ -668,6 +673,12 @@ func recoverDeadPrimary(ctx context.Context, analysisEntry inst.ReplicationAnaly
 		return false, nil, err
 	}
 	log.Infof("Analysis: %v, deadprimary %+v", analysisEntry.Analysis, analysisEntry.AnalyzedInstanceKey)
+	var promotedReplica *inst.Instance
+	// This has to be done in the end; whether successful or not, we should mark that the recovery is done.
+	// So that after the active period passes, we are able to run other recoveries.
+	defer func() {
+		resolveRecovery(topologyRecovery, promotedReplica)
```

Comment:
@GuptaManan100 the way `golang`'s `defer` works is that the value of `promotedReplica` is evaluated _right now_, and so it is `nil`, even if you later change it to something else.
You can use `*(&promotedReplica)` to get the actual value you want.

Reply:
@GuptaManan100 is correct, and the problem is only when running a `defer` statement. The wrapping inside a function is fine :+1: